docs/strategy-advanced.md

```diff
@@ -8,11 +8,111 @@ If you're just getting started, please be familiar with the methods described in
+
+!!! Tip "Trailing stoploss"
+    It's recommended to disable `trailing_stop` when using custom stoploss values. Both can work in tandem, but you might encounter the trailing stop to move the price higher while your custom function would not want this, causing conflicting behavior.
+
+### Custom stoploss examples
+
+The next section will show some examples on what's possible with the custom stoploss function.
+Of course, many more things are possible, and all examples can be combined at will.
+
+#### Time based trailing stop
+
+Use the initial stoploss for the first 60 minutes, after this change to 10% trailing stoploss, and after 2 hours (120 minutes) we use a 5% trailing stoploss.
+
+``` python
+    use_custom_stoploss = True
+
+    def custom_stoploss(self, pair: str, trade: Trade, current_time: datetime, current_rate: float,
+                        current_profit: float, **kwargs) -> float:
+        # TODO: Add full docstring here
+
+        # Make sure you have the longest interval first - these conditions are evaluated from top to bottom.
+        if current_time - timedelta(minutes=120) > trade.open_date:
+            return -0.05
+        elif current_time - timedelta(minutes=60) > trade.open_date:
+            return -0.10
+        return 1
+```
+
+#### Different stoploss per pair
+
+Use a different stoploss depending on the pair.
+In this example, we'll trail the highest price with 10% trailing stoploss for `ETH/BTC` and `XRP/BTC`, with 5% trailing stoploss for `LTC/BTC` and with 15% for all other pairs.
+
+``` python
+    use_custom_stoploss = True
+
+    def custom_stoploss(self, pair: str, trade: Trade, current_time: datetime, current_rate: float,
+                        current_profit: float, **kwargs) -> float:
+        # TODO: Add full docstring here
+
+        if pair in ('ETH/BTC', 'XRP/BTC'):
+            return -0.10
+        elif pair in ('LTC/BTC'):
+            return -0.05
+        return -0.15
+```
+
+#### Absolute stoploss
+
+The below example sets absolute profit levels based on the current profit.
+
+* Use the regular stoploss until 20% profit is reached
+* Once profit is > 40%, stoploss will be at 25%, locking in at least 25% of the profit.
+* Once profit is > 25% - stoploss will be 15%.
+* Once profit is > 20% - stoploss will be set to 7%.
+
+``` python
+    use_custom_stoploss = True
+
+    def custom_stoploss(self, pair: str, trade: Trade, current_time: datetime, current_rate: float,
+                        current_profit: float, **kwargs) -> float:
+        # TODO: Add full docstring here
+
+        # Calculate as `-desired_stop_from_open + current_profit` to get the distance between current_profit and initial price
+        if current_profit > 0.40:
+            return (-0.25 + current_profit)
+        if current_profit > 0.25:
+            return (-0.15 + current_profit)
+        if current_profit > 0.20:
+            return (-0.7 + current_profit)
```

Comment:
If I'm following, this is an example of step-wise varying trailing stoploss. 

For this to address #1193 's proportional trailing stoploss, would it be something like the following? (It uses values from @yaotian's original message.)

``` python
    use_custom_stoploss = True
    trailing_stop_enabler = 0.10
    profit_reduce_checker = 0.50

    def threshold_reached(self, pair: str, trade: Trade, current_rate: float) -> boolean:
        already_reached = check_new_column_in_db_for_this_trade()
        if (already_reached):
            return true
        if (current_rate > trailing_stop_enabler):
            set_new_column_in_db_for_this_trade(true)
            return true
        return false

    def custom_stoploss(self, pair: str, trade: Trade, current_time: datetime, current_rate: float,
                        current_profit: float, **kwargs) -> float:
        if (threshold_reached()):
            return (current_profit - profit_reduce_checker )
        return 1
```

My personal opinion on alternate names for those config values is:
- `trailing_stop_enabler` --> `proportial_trailing_stoploss_threshold`
- `profit_reduce_checker` --> `proportial_trailing_stoploss_ratio`

I suppose adding a new column and setting/getting its values in the `trades` table involves changes to model.py and its tests.

... Or is there a simpler way?

Reply:
i don't see what the new column would track.

`custom_stoploss()` will only allow you to move the stoploss upwards, so this "tracking" is done for you.

assume you bought at 100$ - and initial stop is at 90$...
now you move stop to 95$ - which means you can never place a new stop < 95$.
Whatever you return - if it's below the latest stop, it'll be ignored (you return a ratio relative to current profit - but the stoploss is in the end an absolute value) 

If it's higher tan the existing stop (say it comes out at 95.5$ - then it'll be replaced).

---

The simplest case is to return a static value (say 5%) - with profits for the trade raising, you also raise stoploss. If the profit (current price) falls down - the stoploss will remain where it is, eventually triggering if price falls by 5% from the highest reached price.

This would be identical to a regular trailing_stop with 5% ... but you are now given the possibility to return different values based on "whatever".

Reply:
The example you should be looking at for #1193 is the `Absolute stoploss` one ... while it doesn't use the same values / numbers, it uses a similar approach.

Reply:
Aha... yes, with the only-rising moving custom stoploss column, I agree that a `threshold_reached` column wouldn't be needed, and that my approach could be refactored to utilize the custom stoploss column.

I noticed I made a critical omission regarding my example above being _proportional_. I just edited it from:

`return (current_profit - profit_reduce_checker)`

to:

`return (current_profit - profit_reduce_checker * current_profit)`

Summarizing, I don't currently think there's an issue with the code. Though, I may still tweak and propose and additional example to detail how to address #1193 as I understand it.

Reply:
my point is - the "trailing_stop enabler" the issue talks about, is really just an `if current_profit > 0.10:` ... 
(you only want to start with the proportional stoploss once you reached X% profit - basically it would mimic the behaviour of `trailing_stop_positive_offset` ).

More examples are never wrong - you can make a PR against the `feat/stoploss_custom`  branch - so it'll all be merged at once (or add additional examples later, too).

As always, it's examples ... which means it's there to give you ideas, inspire you - but will never cover all ideas and usecases ... :D

Reply:
💯 No blockers from me, then. Just won't approve until I get to review the rest more thoroughly.

When I get to through together another example 🤞 I'll PR it to this branch of it hasn't been merged yet.

Reply:
i don't intend to merge this "soon", i suspect it'll be a PR at least until friday, more likely until saturday or sunday,a s i don't intend to include this in the december release...